ddl/placement/bundle.go

```diff
@@ -167,60 +151,31 @@ func NewBundleFromSugarOptions(options *model.PlacementSettings) (*Bundle, error
 	}
 	schedule := options.Schedule
 
-	var constraints Constraints
-	var err error
+	// regions must include the primary
+	sort.Strings(regions)
+	primaryIndex := sort.SearchStrings(regions, primaryRegion)
+	if primaryIndex >= len(regions) || regions[primaryIndex] != primaryRegion {
+		return nil, fmt.Errorf("%w: primary region must be included in regions", ErrInvalidPlacementOptions)
+	}
+
+	var Rules []*Rule
 
-	Rules := []*Rule{}
 	switch strings.ToLower(schedule) {
 	case "", "even":
-		constraints, err = NewConstraints([]string{fmt.Sprintf("+region=%s", primaryRegion)})
-		if err != nil {
-			return nil, fmt.Errorf("%w: invalid PrimaryRegion '%s'", err, primaryRegion)
-		}
-		Rules = append(Rules, NewRule(Leader, 1, constraints))
+		Rules = append(Rules, NewRule(Leader, 1, NewConstraintsDirect(NewConstraintDirect("region", In, primaryRegion))))
```

Comment:
I'm thinking about when user will set `SCHEDULE="event"`. On reason I can think of is that user may want to keep the service still be available event if one region crashes. However, the current algorithm can not make it work in some corner cases. For example, user specifies the placement as `PRIMARY_REGION="r1" REGIONS="r1,r2,r3" FOLLOWERS=2`. To achieve the above goal, the peer count in each regions should be `r1: 1, r2: 1, r3: 1` (the peer in r1 is leader). But in fact it may have a probability that r1 has 2 peers (1 leader and one follower) because pd may still assign 1 of the 2 followers to it. If region r1 crashes, the service will be unavailable.

Another way to process "even" option is to create a rule with a role: `Voter`, count: `Ceil((followers + 1 ) / regionsCount)` and constraint: `regions in [primary region]`. And another another rule with a role `Follower`, count: `left peers not in primary region` and constraint: `regions in [regions except primary region]`

Reply:
> because pd may still assign 1 of the 2 followers to it

It is true.

> Another way to process "even" option is to create a rule with a role...

I've thought about it. The thing is when there is the remainder, e.g. `followers = 1 (mod len(regions))`, it will never schedule to the primary. So it kinda unfair for the primary.

But yes, it may be better for most cases. I've changed.